### DISCUSSION TITLE
ITwin access token increase expiresAt more then one hour [Evercam]

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#2904] [Q&A] [answered]
Using the [https://ims.bentley.com/connect/token](https://ims.bentley.com/connect/token) endpoint we managed to request ITwin access token.
But it return a token valid for only 1 hour, is there a way to request a specific expiry date

**Comment by laurynasr** [comment, marked as answer]
There is no possibility to get access_token with longer lifetime. However, depending on the type of application you're building, you have other options
1. If your application authenticates users with IMS (i.e. SPA, Mobile/desktop client or WebApp), you can request offline_access scope and then [use refresh_token grant type to get a new access_token](https://developer.bentley.com/apis/overview/authorization/#usingrefreshtokens) when old one expires. The refresh_token is valid for 30 days.
2. If your application is a backend service (Service application type), the communication with iTwin Platform APIs is done using a service identity and you don't use user authentication at all, so you can simply [issue a new token request](https://developer.bentley.com/apis/overview/authorization/#authorizingservicemachinetomachine) when old token expires.

All application authorization scenarios are documented in the [Authorization](https://developer.bentley.com/apis/overview/authorization/) page.
